src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -86,19 +86,36 @@ const Map<wifi_iface_state, WiFiManager::StationStatus, 10>
                               { WIFI_STATE_COMPLETED, WiFiManager::StationStatus::FULLY_PROVISIONED } });
 
 const Map<uint32_t, WiFiManager::NetEventHandler, 4>
-    WiFiManager::sEventHandlerMap({ { NET_EVENT_WIFI_SCAN_RESULT, WiFiManager::ScanResultClbk },
-                                    { NET_EVENT_WIFI_SCAN_DONE, WiFiManager::ScanDoneClbk },
-                                    { NET_EVENT_WIFI_CONNECT_RESULT, WiFiManager::ConnectClbk },
-                                    { NET_EVENT_WIFI_DISCONNECT_RESULT, WiFiManager::DisconnectClbk } });
+    WiFiManager::sEventHandlerMap({ { NET_EVENT_WIFI_SCAN_RESULT, WiFiManager::ScanResultHandler },
+                                    { NET_EVENT_WIFI_SCAN_DONE, WiFiManager::ScanDoneHandler },
+                                    { NET_EVENT_WIFI_CONNECT_RESULT, WiFiManager::ConnectHandler },
+                                    { NET_EVENT_WIFI_DISCONNECT_RESULT, WiFiManager::DisconnectHandler } });
 
-void WiFiManager::WifiMgmtEventHandler(NetEventCallback * cb, uint32_t mgmtEvent, struct net_if * iface)
+void WiFiManager::WifiMgmtEventHandler(NetEventCallback * cb, uint32_t mgmtEvent, NetInterfacePtr iface)
 {
     if (0 == strcmp(iface->if_dev->dev->name, "wlan0"))
     {
-        DeviceLayer::SystemLayer().ScheduleLambda([cb, mgmtEvent] { sEventHandlerMap[mgmtEvent](cb); });
+        NetworkEventData * eventData = new NetworkEventData;
+        VerifyOrReturn(nullptr != eventData);
+        uint8_t * info = new uint8_t[cb->info_length];
+        VerifyOrReturn(nullptr != info);
```

Comment:
Please use `Platform::UniquePtr` to store the pointer and call `release()` on it only if the `ScheduleWork` or `ScheduleLambda` and everything else succeeds (not to leak memory on failure).

Reply:
My first approach was to pass the ownership to the handler by moving the `std::unique_ptr` with data to the `ScheduleLambda` (in the capture scope)  - that would be a good semantics here IMO but it appears that the API expects trivially copyable lambda, so we cannot capture by move.

So how can we actually use the `Platform::UniquePtr` here? By passing underlying raw pointer directly to the lambda? Don't we need two allocations here anyway meaning that the `mData` inside `NetworkEventData` should be a unique_ptr as well?

Reply:
My point is that we should use the unique pointer only to protect the buffer from leaking. Once the buffer is passed to another thread we can release the unique pointer.

Reply:
Ok, done.